File: combinatorics/resource/ST/project_5/bai_toan_4/mutil_graph/amap_to_eal.py
def amap_to_eal(amap_entries):
    # Vi AMap cung dang (u, v, w) nen chi can tra lai danh sach nay
    return amap_entries

# Nhap tu ban phim
n = int(input("Nhap so dinh: "))
k = int(input("Nhap so bo (u v w): "))

print("Nhap cac bo (u v w):")
amap_entries = []
for _ in range(k):
    u, v, w = map(int, input().split())
    amap_entries.append((u, v, w))

eal = amap_to_eal(amap_entries)

print("Danh sach ke mo rong (EAL):")
for u, v, w in eal:
    print(f"{u} {v} {w}")

File: combinatorics/resource/ST/project_5/bai_toan_4/mutil_graph/amap_to_eal.cpp
#include <iostream>
#include <vector>
#include <tuple>

using namespace std;

// Vi AMap va EAL deu la danh sach (u, v, w) nen tra ve truc tiep
vector<tuple<int, int, int>> amapToEal(const vector<tuple<int, int, int>>& amap_entries) {
    return amap_entries;
}

int main() {
    int n, k;
    cout << "Nhap so dinh: ";
    cin >> n;
    cout << "Nhap so bo (u v w): ";
    cin >> k;

    vector<tuple<int, int, int>> amap_entries;

    cout << "Nhap cac bo (u v w):\n";
    for (int i = 0; i < k; ++i) {
        int u, v, w;
        cin >> u >> v >> w;
        amap_entries.emplace_back(u, v, w);
    }

    auto eal = amapToEal(amap_entries);

    cout << "Danh sach ke mo rong (EAL):\n";
    for (auto [u, v, w] : eal) {
        cout << u << " " << v << " " << w << "\n";
    }

    return 0;
}